packages/protocol/contracts/ZkAsset/ZkAssetMintable.sol

```diff
@@ -0,0 +1,83 @@
+
+
+pragma solidity >=0.5.0 <0.6.0;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./ZkAsset.sol";
+import "../ACE/ACE.sol";
+import "../libs/LibEIP712.sol";
+import "../libs/ProofUtils.sol";
+
+
+
+contract ZkAssetMintable is ZkAsset {
+    event UpdateTotalMinted(bytes32 noteHash, bytes noteData);
+
+    constructor(
+        address _aceAddress,
+        address _linkedTokenAddress,
+        uint256 _scalingFactor,
+        bool _canAdjustSupply,
+        bool _canConvert
+    ) public ZkAsset(
+        _aceAddress,
+        _linkedTokenAddress,
+        _scalingFactor,     
+        _canAdjustSupply,
+        _canConvert
+    ) {
+    }
+
+    function confidentialMint(uint24 _proof, bytes calldata _proofData) external {
+        require(_proofData.length != 0, "proof invalid");
```

Comment:
See burn contract for permissioning notes

Reply:
Done